### DISCUSSION TITLE
GPU Memory-Usage, full finetuning vs LoRA vs LoRA + CPU Offloading

### DISCUSSION THREAD

**Discussion by EthanChen1234** [huggingface/peft#1804] [Q&A]
![image](https://github.com/huggingface/peft/assets/45474996/2bcead39-534e-4661-8b0c-5983f6c9edd6)

It's a great work!

hi, how to calculate the GPU Memory in different methods?

In addition, using LoRA method, the base parameter will go through forward pass, will the base parameter go through backward pass?

**Comment by younesbelkada** [comment]
Hi ! 
Thanks ! 
One way to calculate effectively memory stats is to call: `torch.cuda.reset_peak_memory_stats(device)` at the beginning of your script then call: `max_memory = torch.cuda.max_memory_allocated(device)`

> In addition, using LoRA method, the base parameter will go through forward pass, will the base parameter go through backward pass?

Since backward pass is based on the chain rule, I would say so, however we only calculate gradients on the LoRA parameters

**Reply by EthanChen1234** [reply]
ok, thanks
